framework/include/base/SubProblem.h

```diff
@@ -368,6 +400,12 @@ class SubProblem : public Problem
   /* This needs to remain <unsigned int> for threading purposes */
   std::vector<unsigned int> _has_active_elemental_moose_variables;
 
+  /// Set of material property ids that determine whether materials get reinited
+  std::vector<std::set<unsigned int>> _active_material_property_ids;
+
+  /// True if _active_material_property_ids is not empty
+  std::vector<unsigned int> _has_active_material_property_ids;
```

Comment:
Easy and clean to just remove the second vector entirely though.  I'm in favor of doing it well the first time - we've all experienced the longevity of temporary solutions ;)